src/javascript/app/pages/trade/digit_trade.js

```diff
@@ -0,0 +1,94 @@
+        tick_count = 1;
+        contract   = proposal_open_contract;
+
+        $container = $(`#${id_render}`);
+        $container
+            .addClass('normal-font')
+            .html($('<h5 />', { text: contract.display_name, class: 'center-text' }))
+            .append($('<div />', { class: 'gr-6 gr-centered' })
+                .append($('<div />', { class: 'gr-row', id: 'table_digits' })
+                    .append($('<strong />', { class: 'gr-3', text: localize('Tick') }))
+                    .append($('<strong />', { class: 'gr-3', text: localize('Spot') }))
+                    .append($('<strong />', { class: 'gr-6', text: localize('Spot Time (GMT)') }))))
+            .append($('<div />', { class: 'digit-ticker invisible', id: 'digit-ticker-container' }));
+
+        const request     = {
+            ticks_history: contract.underlying,
+            start        : contract.date_start,
+        };
+        if (contract.current_spot_time < contract.date_expiry) {
+            request.subscribe = 1;
+            request.end       = 'latest';
+        } else {
+            request.end = contract.date_expiry;
+        }
+        DigitTicker.init('digit-ticker-container', contract);
+
+        BinarySocket.send(request, { callback: update });
+    };
+
+    const updateTable = (spot, time) => {
+        $container
+            .find('#table_digits')
+            .append($('<p />', { class: 'gr-3', text: tick_count }))
+            .append($('<p />', { class: 'gr-3 gray', html: tick_count === contract.tick_count ? `${spot.slice(0, spot.length - 1)}<strong>${spot.slice(-1)}</strong>` : spot }))
+            .append($('<p />', { class: 'gr-6 gray digit-spot-time no-underline', text: moment(+time * 1000).utc().format('YYYY-MM-DD HH:mm:ss') }));
+
+        DigitTicker.update(
+            tick_count,
+            spot.slice(-1),
+            contract
+        );
+    };
+
+    const update = (response) => {
+        if (!$container.is(':visible') || !response || (!response.tick && !response.history)) {
+            return;
+        }
+
+        if (getPropertyValue(response, ['tick', 'id']) && document.getElementById('sell_content_wrapper')) {
+            ViewPopupUI.storeSubscriptionID(response.tick.id);
+        }
+
+        if (response.history) {
+            response.history.times.some((time, idx) => {
+                if (+time >= +contract.entry_tick_time) {
+                    updateTable(response.history.prices[idx], time);
+                    tick_count += 1;
+                }
+                return tick_count > contract.tick_count;
+            });
+        } else if (response.tick) {
+            updateTable(response.tick.quote, response.tick.epoch);
+            tick_count += 1;
+        }
+        showLocalTimeOnHover('.digit-spot-time');
+    };
+
+    const end = (proposal_open_contract) => {
+        contract = proposal_open_contract;
+        // here we need to highlight won/lost
```

Comment:
I left this so we could pass the final status to DigitTicker, assuming that view popup is opened while contract is streaming and we don't know if won/lost, since contract gets past only on init, so it needs to be passed again once it ends.

Reply:
Calling the end is not required. I removed it.